org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java

```diff
@@ -84,6 +87,10 @@ public Image getImage(ImageDescriptor descriptor) {
 		if (result != null) {
 			return result;
 		}
+		if (!displayDisposedRunnableInitialized  ) {
```

Comment:
have no idea on how to do this "nicer"
also dont know if we need the bundle listener at all.

Reply:
am also not sure if this will leak the helper itself / the injector to the display.

Reply:
Also not sure you can do it nicer, other than factoring it out in a method as mentioned below. As far as I can see the relevant methods will only ever be called from the UI thread so a regular boolean should be fine.

I'm guessing the helper will indeed be leaked and the object will be kept around even when it's disposed early.

@szarnekow Any idea about injector leaks?